repose-aggregator/components/filters/valkyrie-authorization/src/main/scala/org/openrepose/filters/valkyrieauthorization/ValkyrieAuthorizationFilter.scala

```diff
@@ -96,12 +117,31 @@ class ValkyrieAuthorizationFilter @Inject()(configurationService: ConfigurationS
       case ResponseResult(code, message) =>
         mutableHttpResponse.sendError(code, message)
     }
+
+    //weep for what we must do with this flaming pile, note above where it was wrapped pointlessly just to break the chain
+    mutableHttpResponse.writeHeadersToResponse()
+    mutableHttpResponse.commitBufferToServletOutputStream()
   }
 
   def cacheKey(transformedTenant: String, contactId: String): String = {
     transformedTenant + contactId
   }
 
+  def nonAuthorizedPath(url: => String): Boolean = {
+    val path: String = new URL(url).getPath
+    lazy val onResourceList: Boolean = Option(configuration.getCollectionResources)
+      .getOrElse(new CollectionResources)
+      .getResource.asScala.exists { resource =>
+      resource.getPathRegex.r.findFirstIn(path).isDefined
```

Comment:
As an added benefit, using patter.matcher(...).matches attempts to match the entire region rather than a section of it. This is likely the intended behavior that .findFirstIn was not providing.

Consider the following cases:
regex | path | findFirstIn | pattern.matcher.matches
------- | ------ | ------------- | ---------------------------------
/dog | /dog | true | true
/dog | /dog/cat | true | false
/dog | /cat/dog | true | false
^/dog$ | /dog | true | true
^/dog$ | /dog/cat | false | false
^/dog$ | /cat/dog | false | false